### PR TITLE
Add JSON documentation generator & update to Angular v17

### LINKED ISSUE #300: Create a JSON generator (to generate json files from ts annotations)

### LINKED ISSUE #301: Update the docs app to consume generated json files

### PR DESCRIPTION
**Pull request by lukasmatta**
This PR introduces JSON generator, that can create JSON files containing documentation for the components/services/directives based on the annotations in the files themselves. It is more or less took from the [PrimeNg lib](https://github.com/primefaces/primeng/blob/master/api-generator/build-apidoc.ts) with some modifications to serve our purposes.

I've also updated all the pages in the composition app to make use of those JSON files. All the routes are now lazily loaded. There is also a Github Action to build the composition app and upload it to Github pages.

Angular in the library is now updated to version 17.

Also updated Node version to 20 in the workflows as the v16 is no longer supported. Added option to skip automatic version bump before publishing by including `#SkipVersionBump` in the squash commit, to be able to publish new major/patch version without increasing the minor version.

Closes #300, Closes #301

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,39 +5,51 @@
     "ng": "ng",
     "start": "ng serve",
     "build": "ng build --configuration=production --project=cps-ui-kit",
+    "build:documentation": "ng build --configuration=production --project=composition --base-href '/cps-shared-ui/'",
     "watch": "ng build cps-ui-kit --watch --configuration development",
     "test": "npm run cypress:run",
     "lint": "eslint \"**/*.ts\"",
     "cypress:open": "cypress open --component --browser chrome",
-    "cypress:run": "cypress run --component"
+    "cypress:run": "cypress run --component",
```
Comment (fateeand) [inline package.json:13]: these are not needed I assume:
```
"dev:ssr": "ng run composition:serve-ssr",
 "serve:ssr": "node dist/composition/server/main.js",
 "build:ssr": "ng build && ng run composition:server",
 "prerender": "ng run composition:prerender"
```

File: package.json [hunk 2]
```diff
@@ -5,39 +5,51 @@
...
+    "@angular/animations": "^17.1.0",
+    "@angular/common": "^17.1.0",
+    "@angular/compiler": "^17.1.0",
+    "@angular/core": "^17.1.0",
+    "@angular/forms": "^17.1.0",
+    "@angular/platform-browser": "^17.1.0",
+    "@angular/platform-browser-dynamic": "^17.1.0",
+    "@angular/platform-server": "^17.1.0",
+    "@angular/router": "^17.1.0",
+    "@angular/ssr": "^17.1.0",
```
Comment (fateeand) [inline package.json:31]: @angular/ssr is not needed as well

File: package.json [hunk 3]
```diff
@@ -5,39 +5,51 @@
...
     "@types/lodash-es": "^4.17.7",
+    "express": "^4.18.2",
     "lodash-es": "^4.17.21",
-    "primeng": "^15.4.1",
+    "primeng": "^17.3.2",
```
Comment (fateeand) [inline package.json:35]: pls update:
angular to 17.1.1
primeng to 17.4.0
@types/lodash-es to 4.17.12
rxjs to 7.8.1
tslib to 2.6.2
zone.js to 0.14.3
eslint to 8.52.0
prettier to 3.2.4
ng-packagr to 17.1.2
and so on :)

Comment (lukasmatta) [reply package.json]: Updated

File: package.json (hunk 1 above)
Comment (lukasmatta) [reply package.json:13]: it's preparation for using prerendering is it necessary to remove it?

Comment (fateeand) [reply package.json:13]: Ok, let's keep it

File: projects/cps-ui-kit/package.json [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
-  "version": "0.164.0",
+  "version": "17.0.0",
   "peerDependencies": {
-    "@angular/common": "15 - 16",
-    "@angular/core": "15 - 16",
-    "@angular/forms": "15 - 16",
+    "@angular/common": "17",
+    "@angular/core": "17",
+    "@angular/forms": "17",
     "rxjs": "^7.8.0",
```
Comment (fateeand) [inline projects/cps-ui-kit/package.json:8]: please update peerDependencies as well

**Review by fateeand — APPROVED** [review]

**Pull request merged**